### ISSUE TITLE
Panel decorator from derived to base class

### ISSUE THREAD

**Issue by Estrusco** [volkanceylan/Serenity#1888]
Good morning @volkanceylan 

I have an normal dialog

```cs
    @Serenity.Decorators.registerClass()
    @Serenity.Decorators.responsive()
    @Serenity.Decorators.maximizable()
    export class XYZDialog extends Serenity.EntityDialog<XYZRow, Service.ObjectRequest> {

    [...]
```

and a derived panel 

```cs
   @Serenity.Decorators.panel()
    export class XYZDialogPanel extends XYZDialog  {
       constructor(option: Service.ObjectRequest) {
            super(option);

            this.element.addClass('flex-layout');
        }
   }
```

This becouse I need the same dialog in a panel and a normal dialog in two different situation.

The problem is that when I use the normal dialog, in Serenity.CoreLib.js

```js
        function TemplatedDialog(options) {
            var _this = _super.call(this, Q.newBodyDiv(), options) || this;
            _this.isPanel = ss.getAttributes(ss.getInstanceType(_this), Serenity.PanelAttribute, true).length > 0;
```

_this.isPanel is true. Seems that if a derived class as the Panel decorator, also the base class get the Panel decorator.

I have make a workaround adding a _this.setIsPanel(_this.isPanel) function that I override in the XYZDialog class.

 TemplateDialog.ts in Serenity.TypeScript.CoreLib that I have modified:
```ts
        constructor(options?: TOptions) {
            super(Q.newBodyDiv(), options);

            this.isPanel = (ss as any).getAttributes((ss as any).getInstanceType(this),
                Serenity.PanelAttribute, true).length > 0;

            // ADDED
            this.isPanel = this.setIsPanel(this.isPanel);

            if (!this.isPanel) {
                this.initDialog();
            }

            this.initValidator();
            this.initTabs();
            this.initToolbar();
        }

        // ADDED
        protected setIsPanel(isPanel: boolean): boolean {
            return isPanel;
        }
```

in dialog

```ts
        protected setIsPanel(isPanel: boolean): boolean {
            return false;
        }
```

Can I send you a pull request with this change?

**Comment by volkanceylan** [comment]
This will be fixed in 2.9.7, it's a bug related to __extends helper of TypeScript and Saltaralle type system not being compatible.

**Comment by Estrusco** [comment]
Thank you very much @volkanceylan
